internal/flink/command_shell.go

```diff
@@ -117,12 +120,23 @@ func (c *command) startFlinkSqlClient(prerunner pcmd.PreRunner, cmd *cobra.Comma
 		return err
 	}
 	if identityPool == "" {
-		if c.Context.GetCurrentIdentityPool() == "" {
-			return errors.NewErrorWithSuggestions("no identity pool set", "Set a persistent identity pool with `confluent iam pool use` or pass the `--identity-pool` flag.")
-		}
 		identityPool = c.Context.GetCurrentIdentityPool()
 	}
 
+	serviceAccount, err := cmd.Flags().GetString("service-account")
+	if err != nil {
+		return err
+	}
+	if serviceAccount == "" {
+		serviceAccount = c.Context.GetCurrentServiceAccount()
+	}
+
+	if serviceAccount == "" && identityPool == "" {
+		output.ErrPrintln("Warning: no service account provided. To ensure that your statements run continuously, " +
+			"switch to using a service account instead of your user identity by providing a service account with `confluent iam service-account use` or `--service-account`. " +
```

Comment:
Previous text was a bit redundant.
```suggestion
			"switch to using a service account instead of your user identity with `confluent iam service-account use` or `--service-account`. " +
```